### ISSUE TITLE
mockServerClient verification fails for slightly different URLs on 3.10.7

### ISSUE THREAD

**Issue by felipecao** [jamesdbloom/mockserver#336]
I've put up an example project to demo this issue: https://github.com/felipecao/mockserver-3.10.7-bug

This project has just a test class: `HttpTest`

If you run the tests, you'll notice that the second test fails for apparently no reason. The fix I found was to downgrade `mockserver-netty` from `3.10.7` to `3.10.4`.

Am I doing anything wrong here? Is my test written wrongfully? Should I be doing the verification somehow differently?

Thanks!

**Comment by jamesdbloom** [comment]
Making the socket dynamic will fix the issue, as follows:  

```java
@Rule
public MockServerRule mockServerRule = new MockServerRule(this);
```

The port can then be retrieved dynamically from the MockServerRule as follows:

```java
mockServerRule.getPort()
```

The problem is not related to any MockServer matching bug.  The actual cause of the problem is due to the socket connection from jax-rs being re-used across clients so the second test ends up sending requests to the MockServer instance configured from the previous test, as the second instance of the jax-rs client re-uses the socket connection from the previous client.  I should be able to improve MockServer to resolve this issue and work around this behaviour of jax-rs.  

The reason this problem started to appear in version 3.10.7 is because I improved the way MockServer shut down and how the client detects that it has shutdown.  This improvement increased the speed with which the MockServer shutdown so the jax-rs client didn't timeout the socket connect.

**Comment by jamesdbloom** [comment]
This issue should be complete resolved in 3.10.8-SNAPSHOT, so I'm closing this ticket.
